sktime/classification/dictionary_based/_boss.py

```diff
@@ -101,37 +98,23 @@ class BOSSEnsemble(BaseClassifier):
     def __init__(self,
                  randomised_ensemble=False,
                  n_parameter_samples=250,
-                 random_state=None,
                  threshold=0.92,
-                 max_ensemble_size=None,
+                 max_ensemble_size=500,
                  max_win_len_prop=1,
                  time_limit=0.0,
-                 word_lengths=None,
                  alphabet_size=4,
                  min_window=10,
-                 norm_options=None
+                 random_state=None
                  ):
-        if word_lengths is None:
-            word_lengths = [16, 14, 12, 10, 8]
-        if norm_options is None:
-            norm_options = [True, False]
-
-        if max_ensemble_size is None:
-            if randomised_ensemble:
-                max_ensemble_size = 50
-            else:
-                max_ensemble_size = 500
-
         self.randomised_ensemble = randomised_ensemble
         self.n_parameter_samples = n_parameter_samples
-        self.random_state = random_state
-        random.seed(random_state)
         self.threshold = threshold
         self.max_ensemble_size = max_ensemble_size
         self.max_win_len_prop = max_win_len_prop
         self.time_limit = time_limit
+        self.random_state = random_state
+        random.seed(random_state)
```

Comment:
Could you use scikit-learn's [`check_random_state`](https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_random_state.html) function in and wherever you need it rather than in the constructor? The function returns a random state object, so you can use that directly to use Numpy's random functionality, e.g.

```python
rng = check_random_state(self.random_state)
subsample = rng.randint(self.n_instances, size=subsample_size)
```

Reply:
I have done this for both of the BOSS classifiers, alongside some mandatory style stuff which was missed initially.